### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by AnasHagras**

### CODE REVIEW

File: albums/models.py [hunk 1]
```diff
@@ -21,4 +23,19 @@ class Album(TimeStampedModel):
...
+        return self.name
+    
+class Song(TimeStampedModel):
+    name = models.CharField(max_length = 50, default="")
+    song_image = models.ImageField(upload_to='songs/photos')
+    song_thumbnail = ImageSpecField(source='song_image',
+                                      processors=[ResizeToFill(100, 50)],
+                                      format='JPEG',
+                                      options={'quality': 100})
+    song_audio = models.FileField(upload_to="songs/files")
```
Comment (yousefokasha12) [inline albums/models.py:35]: you didn't add file extension validator:
```python 
audio = models.FileField(blank=False, null=False,
                             upload_to='static/albums/audios',
                             help_text='.wav, .mp3',
                             validators=[FileExtensionValidator(allowed_extensions=['mp3', 'wav'])])
```

File: artists/views.py [hunk 2]
```diff
@@ -2,44 +2,69 @@
...
 
 from django.core.exceptions import ValidationError
 
+from django.views import View
+
 from .models import Artist
 
+from django.http import HttpResponse
+
+class Index(View):
```
Comment (yousefokasha12) [inline artists/views.py:14]: this view could be minimized into this:
```python
class Index(ListView):
    queryset = Artist.objects.prefetch_related('album_set')
    context_object_name = 'artists_list'
    template_name = 'artists/index.html'
```

File: artists/views.py [hunk 3]
```diff
@@ -2,44 +2,69 @@
...
+            # 'albums' : Artist.objects.filter(Q(album__is_approved=True)).annotate(count = Count('album')).order_by('-count')
+        }
+        return data 
+    def post (self,request,*args,**kwargs):
+        return render(request,"artists/index.html",self.getData(data = kwargs['data']['msg']))
+         
+    def get(self,request,*args,**kwargs):
+        return render(request,"artists/index.html",self.getData(data = ''))
+
+class Create(View):
```
Comment (yousefokasha12) [inline artists/views.py:36]:
```python
class Create(LoginRequiredMixin, CreateView):
    model = Artist
    fields = ['stage_name', 'social_link']
    template_name = 'artists/create.html'
```

File: albums/models.py (hunk 1 above)
Comment (AnasHagras) [reply albums/models.py:35]: okay thanks

File: artists/views.py (hunk 2 above)
Comment (AnasHagras) [reply artists/views.py:14]: there are more operations than getting all artists , what prefetch do ?

File: artists/views.py (hunk 3 above)
Comment (AnasHagras) [reply artists/views.py:36]: what is the point of using this ? , i used different way but , what's better ?

**Pull request merged**